plugins/utils/todo-comments.nix

```diff
@@ -1,290 +1,429 @@
+    ]
+    [
+      "highlight"
+      "maxLineLen"
+    ]
+    [
+      "highlight"
+      "exclude"
+    ]
+    [
+      "colors"
+      "error"
+    ]
+    [
+      "colors"
+      "warning"
+    ]
+    [
+      "colors"
+      "info"
+    ]
+    [
+      "colors"
+      "hint"
+    ]
+    [
+      "colors"
+      "default"
+    ]
+    [
+      "colors"
+      "test"
+    ]
+    [
+      "search"
+      "command"
+    ]
+    [
+      "search"
+      "args"
+    ]
+    [
+      "search"
+      "pattern"
+    ]
+  ];
 
-      signs = helpers.defaultNullOpts.mkBool true "Show icons in the signs column.";
-
-      signPriority = helpers.defaultNullOpts.mkInt 8 "Sign priority.";
-
-      keywords =
-        helpers.mkNullOrOption
-          (types.attrsOf (
-            types.submodule {
-              options = {
-                icon = helpers.mkNullOrOption types.str ''
-                  Icon used for the sign, and in search results.
-                '';
-
-                color = helpers.mkNullOrOption types.str ''
-                  Can be a hex color, or a named color.
-                '';
-
-                alt = helpers.mkNullOrOption (types.listOf types.str) ''
-                  A set of other keywords that all map to this FIX keywords.
-                '';
-
-                signs = helpers.mkNullOrOption types.bool ''
-                  Configure signs for some keywords individually.
-                '';
-              };
-            }
-          ))
-          ''
-            Configurations for keywords to be recognized as todo comments.
-
-            Default:
-            ```nix
-            {
-              FIX = {
-                icon = " "; # Icon used for the sign, and in search results.
-                color = "error"; # Can be a hex color, or a named color.
-                alt = [ "FIXME" "BUG" "FIXIT" "ISSUE" ]; # A set of other keywords that all map to this FIX keywords.
-              };
-              TODO = { icon = " "; color = "info"; };
-              HACK = { icon = " "; color = "warning"; };
-              WARN = { icon = " "; color = "warning"; alt = [ "WARNING" "XXX" ]; };
-              PERF = { icon = " "; alt = [ "OPTIM" "PERFORMANCE" "OPTIMIZE" ]; };
-              NOTE = { icon = " "; color = "hint"; alt = [ "INFO" ]; };
-              TEST = { icon = "⏲ "; color = "test"; alt = [ "TESTING" "PASSED" "FAILED" ]; };
-            };
-            ```
-          '';
+  settingsOptions = {
+    signs = defaultNullOpts.mkBool true "Show icons in the signs column.";
 
-      guiStyle = {
-        fg = helpers.defaultNullOpts.mkStr "NONE" ''
-          The gui style to use for the fg highlight group.
-        '';
+    sign_priority = defaultNullOpts.mkInt 8 "Sign priority.";
 
-        bg = helpers.defaultNullOpts.mkStr "BOLD" ''
-          The gui style to use for the bg highlight group.
-        '';
-      };
+    keywords =
+      defaultNullOpts.mkAttrsOf
+        (types.submodule {
+          options = {
```

Comment:
Should this have a freeform type? Or should we restrict users to only these sub-options?

Reply:
I'm torn on it, these are the kinds of things that are limited to options and it can be nice to have available in documentation. I think it will be easier to remove the maintenance when we figure out the documentation only options.

Reply:
I meant having `freeformType = attrsOf anything` in addition to `options = {}`.

We don't need to do this, if you want to restrict the user to only the sub-options declared, but if you don't want that restriction, we need to define a freeformType.

Reply:
That makes sense, I was thinking it replaced the existing options for some reason.